packages/graphql-playground-react/src/components/EndpointPopup.tsx

```diff
@@ -20,7 +22,37 @@ export interface State {
 
 export default class EndpointPopup extends React.Component<Props, State> {
   checkEndpoint = throttle(() => {
-    if (this.state.endpoint.match(/^https?:\/\/\w+(\.\w+)*(:[0-9]+)?\/?.*$/)) {
+    if (this.state.endpoint.match(/^(https?|wss?):\/\/\w+(\.\w+)*(:[0-9]+)?\/?.*$/)) {
+      if (this.state.endpoint.match(/^(wss?)/)) {
+        const client = createClient({
+          url: this.state.endpoint,
+          retryAttempts:0
+        });
+        const unsubscribe = client.subscribe(
+          {
+            query: `{
+              __schema {
+                queryType {
+                  kind
+                }
+              }
+            }`,
+          },
+          {
+            next: () => {
+              this.setState({ valid: true })
+              unsubscribe()
+            },
+            error: () => {
+              this.setState({ valid: false });
+              unsubscribe()
+            },
+            complete: () =>  {},
+          },
+        );  
```

Comment:
A simpler way to achieve the same check would be by disabling `lazy` mode (connect on instantiation) and check if the client `connected`.

```suggestion
        const client = createClient({
          url: this.state.endpoint,
          retryAttempts: 0,
          lazy: false,
          on: {
            closed: () => this.setState({ valid: false }),
            connected: (socket) => {
              this.setState({ valid: true });
              socket.close(1000); // close after validation
            },
          },
        });
```

Reply:
oof, I forgot to commit this suggestion. @lethot or @enisdenjo can you follow on a PR with this correction?